e2e/config/config.go

```diff
@@ -71,8 +71,23 @@ type Deployer struct {
 	Type string `json:"type"`
 	// Description is a human-readable description of the deployer.
 	Description string `json:"description"`
+	Disapp      Disapp `json:"disapp,omitempty"`
 }
 
+type Disapp struct {
+	// RecipeType is the type of the recipe. Can be one of the following:
+	// - "generate" for the test to generate a recipe based on the workload
+	// - "workload" for the test to use a workload provided recipe
+	// - "internal" for the test to use a ramen internal recipe
+	RecipeType string `json:"recipeType"`
+	// RecipeName is the name of the recipe to use.
+	// If RecipeType is "generate", the generated recipe will use this name.
+	// If RecipeType is "workload", the recipeRef in the DRPC will use this name.
```

Comment:
If the type is workload, the workload will provide the name, so we don't need this option.

Reply:
If the workload created the recipe, it knows eveything about the recipe and can return the details we need.

Reply:
Agreed, done.